diff -Nur drakxtools-14.41.old/perl-install/standalone/finish-install drakxtools-14.41/perl-install/standalone/finish-install
--- drakxtools-14.41.old/perl-install/standalone/finish-install	2013-05-21 13:45:09.008791901 +0200
+++ drakxtools-14.41/perl-install/standalone/finish-install	2013-05-21 13:46:24.703825726 +0200
@@ -183,7 +183,7 @@
 	my $services = {
 		'smb'        => { STATE => 0, NAME => "Samba",    DESCRIPTION => N("Windows-compatible sharing system") },
 #		'mdmonitor'  => { STATE => 0, NAME => "Linux MD", DESCRIPTION => N("Software RAID support") },
-		'openvpn'    => { STATE => 0, NAME => "OpenVPN",  DESCRIPTION => N("Virtual Private Network") },
+#		'openvpn'    => { STATE => 0, NAME => "OpenVPN",  DESCRIPTION => N("Virtual Private Network") },
 		'sshd'       => { STATE => 0, NAME => "sshd",     DESCRIPTION => N("OpenSSH server") },
 		'cups'       => { STATE => 0, NAME => "CUPS",     DESCRIPTION => N("Print server") },
 #		'nfs-server' => { STATE => 0, NAME => "NFS",      DESCRIPTION => N("Network File System server") },
