--- drakxtools-14.22/perl-install/bootloader.pm.0006~	2012-05-25 07:28:47.531215809 +0200
+++ drakxtools-14.22/perl-install/bootloader.pm	2012-05-25 07:33:16.198857003 +0200
@@ -131,15 +131,15 @@ sub mkinitrd {
 }
 
 sub rebuild_initrd {
-    my ($kernel_version, $bootloader, $entry, $initrd) = @_;
+#    my ($kernel_version, $bootloader, $entry, $initrd) = @_;
 
-    my $old = $::prefix . $entry->{initrd} . '.old';
-    unlink $old;
-    rename "$::prefix$initrd", $old;
-    if (!mkinitrd($kernel_version, $bootloader, $entry, $initrd)) {
-	log::l("rebuilding initrd failed, putting back the old one");
-	rename $old, "$::prefix$initrd";
-    }
+#    my $old = $::prefix . $entry->{initrd} . '.old';
+#    unlink $old;
+#    rename "$::prefix$initrd", $old;
+#    if (!mkinitrd($kernel_version, $bootloader, $entry, $initrd)) {
+#	log::l("rebuilding initrd failed, putting back the old one");
+#	rename $old, "$::prefix$initrd";
+#    }
 }
 
 sub remove_boot_splash {
@@ -749,6 +749,10 @@ sub add_kernel {
 
     if (!$b_no_initrd) {
 	$v->{initrd} = mkinitrd($kernel_str->{version}, $bootloader, $v, "/boot/$initrd_long");
+    } else {
+	# we just set up copy of the universal initrd, and we need to add proper info
+	# in bootloader configuration
+	$v->{initrd} = '/boot/initrd.img';
     }
 
     if (!$b_nolink) {
@@ -761,7 +765,7 @@ sub add_kernel {
 	    _do_the_symlink($bootloader, $v->{kernel_or_dev}, $vmlinuz_long);
 	}
 
-	if ($v->{initrd}) {
+	if ($v->{initrd} && !$b_no_initrd) {
 	    $v->{initrd} = '/boot/' . kernel_str2initrd_short($kernel_str);
 	    if (arch() =~ /mips/) {
 		log::l("link $::prefix/boot/$initrd_long -> $::prefix$v->{initrd}");
@@ -1081,15 +1085,15 @@ sub suggest {
 		if_($options{vga_fb}, vga => $options{vga_fb}), #- using framebuffer
 		if_($options{vga_fb} && $options{splash}, append => "splash"),
 		if_($options{quiet}, append => "splash quiet"),
-	       });
+	       }, "", 1);
 
 	if ($options{vga_fb} && $e->{label} eq 'linux') {
-	    add_kernel($bootloader, $kernel, { root => $root, label => 'linux-nonfb' });
+	    add_kernel($bootloader, $kernel, { root => $root, label => 'linux-nonfb' }, "", 1);
 	}
     }
 
     add_kernel($bootloader, $kernels[0],
-	       { root => $root, label => 'failsafe', append => 'failsafe' })
+	       { root => $root, label => 'failsafe', append => 'failsafe' }, "", 1)
       if @kernels;
 
     if (arch() =~ /ppc/) {
--- drakxtools-14.22/perl-install/check_min_sys_requirements.pm.0006~	2012-05-25 06:41:27.000000000 +0200
+++ drakxtools-14.22/perl-install/check_min_sys_requirements.pm	2012-05-25 07:28:47.540215696 +0200
@@ -1,6 +1,11 @@
 package check_min_sys_requirements;
 
+use Exporter;
 
+$VERSION = 1.0;
+@ISA = qw(Exporter);
+
+@EXPORT = qw(&get_live_system_size &check_min_sys_requirements);
 
 use common;
 use utf8;
@@ -54,10 +59,17 @@ sub min_system_requirements {
             if ($parameter eq "hdd") {
                $_[1]= $size;
             }
+            if ($parameter eq "imagesize") {
+               $_[2] = $size;
+            }
          }
     }
 }    
 
+sub get_live_system_size {
+	min_system_requirements($unused1, $unused2, $imagesize);
+	return $imagesize;
+}
 
 sub check_min_sys_requirements {
     my ($in) = @_;
