diff -Nur drakxtools-14.41-old/perl-install/standalone/drakbug drakxtools-14.41/perl-install/standalone/drakbug
--- drakxtools-14.41-old/perl-install/standalone/drakbug	2013-06-17 16:55:12.154147141 +0200
+++ drakxtools-14.41/perl-install/standalone/drakbug	2013-06-17 17:01:02.451271786 +0200
@@ -64,13 +64,13 @@
 $window->{window}->signal_connect("delete_event", sub { ugtk2->exit(0) });
 
 my $mdk_app = { 
-	       N("Mandriva Linux Control Center") => 'drakconf',
+	       N("OpenMandriva Lx Control Center") => 'drakconf',
 	       N("First Time Wizard") => 'drakfw',
 	       N("Synchronization tool") => 'draksync',  
 	       N("Standalone Tools") => 'drakxtools',
 	       "harddrake" => 'harddrake2',
-	       N("Mandriva Online") => 'mdkonline',
-	       N("Mandriva Online") => 'mdkapplet',
+	       N("OpenMandriva Online") => 'mdkonline',
+	       N("OpenMandriva Online") => 'mdkapplet',
 	       N("Remote Control") => 'rfbdrake',
 	       N("Software Manager") => 'rpmdrake',
 	       N("Windows Migration tool") => 'transfugdrake',
@@ -92,11 +92,11 @@
 } else {
     $distrocode = "cooker";
 }
-my $bugzilla_url = 'http://qa.mandriva.com/enter_bug.cgi';
+my $bugzilla_url = 'http://issues.openmandriva.org/enter_bug.cgi';
 my $wizard_name = "Bugzilla";
 
 $table = create_packtable({ col_spacings => 5, row_spacings => 10 },
-		          [ gtknew('Label_Left', text => N("Select Mandriva Tool:")), $comb_app = Gtk2::ComboBox->new_text, $comb_app->set_wrap_width(3) ],
+		          [ gtknew('Label_Left', text => N("Select OpenMandriva Tool:")), $comb_app = Gtk2::ComboBox->new_text, $comb_app->set_wrap_width(3) ],
                           [ gtknew('Label_Left', text => N("or Application Name\n(or Full Path):")), 
 			  gtkpack_(Gtk2::HBox->new(0, 5),
 				   1, $com_app = gtkset_editable(Gtk2::Entry->new, 1), 
@@ -237,7 +237,7 @@
 
 sub report_bug_to_bugzilla() {
     my $p = $package->get_text;
-    my ($product, $version) = $p =~ /^(.*)-([^-]+-[^-]+(mdk|mdv.*))$/; # FIXME: fragile!
+    my ($product, $version) = $p =~ /^(.*)-([^-]+-[^-]+(mdk|mdv|omv.*))$/; # FIXME: fragile!
     my $app = $comb_app->entry->get_text;
     my $_component = $app ?
       if_(member($app, @all_drakxtools), $app) || $mdk_app->{$app} :
@@ -265,7 +265,7 @@
                        ($product || $version ? 'cf_rpmpkg=' . join('-', $product, $version) : ()),
                        'version=' . ($rel_data->{branch} eq 'Devel' ? 'Cooker' : $rel_data->{version}),
                        'component=Core%20Packages',
-                       'classification=Mandriva%20Linux',
+                       'classification=OpenMandriva%20Linux',
                        "rep_platform=$arch",
                        if_($incident,
                            join('', "short_desc=$prog%20",
