diff -Nur drakxtools-14.41.old/perl-install/any.pm drakxtools-14.41/perl-install/any.pm
--- drakxtools-14.41.old/perl-install/any.pm	2013-04-16 17:25:43.000000000 +0400
+++ drakxtools-14.41/perl-install/any.pm	2013-04-18 23:27:24.556754028 +0400
@@ -1074,6 +1074,7 @@
     my ($in) = @_;
     my $ext = $in->isa('interactive::gtk') ? '.html' : '.txt';
     my $separator = $in->isa('interactive::gtk') ? "\n\n" : '';
+    my $lang = (($ENV{'LC_MESSAGES'} =~ m/ru_RU/) ? 'ru' : 'en');
 
     my $release_notes = join($separator, grep { $_ } map {
         if ($::isInstall) {
diff -Nur drakxtools-14.41.old/perl-install/background.pm drakxtools-14.41/perl-install/background.pm
--- drakxtools-14.41.old/perl-install/background.pm	1970-01-01 03:00:00.000000000 +0300
+++ drakxtools-14.41/perl-install/background.pm	2012-10-11 17:26:31.000000000 +0400
@@ -0,0 +1,48 @@
+package background;
+
+use Gtk2;
+
+sub draw_bg_pixbuf($$) {
+    my ($widget, $event) = @_;
+    my $gdk_window = $widget->window;
+    my ($w, $h) = $gdk_window->get_size;
+    if (!defined($::bg_pixbuf_orig)) {
+        eval {
+	    $::bg_pixbuf_orig = Gtk2::Gdk::Pixbuf->new_from_file('/usr/share/mdk/backgrounds/default.png');
+        };
+	if (!$::bg_pixbuf_orig) {
+	    print STDERR "Failed to load image file!\n";
+	    return 0;
+        }
+    }
+    if (!defined($::bg_pixbuf)) {
+	$::bg_pixbuf = $::bg_pixbuf_orig;
+    }
+    my ($pw, $ph) = ($::bg_pixbuf->get_width, $::bg_pixbuf->get_height);
+    if (($w != $pw) or ($h != $ph)) {
+        $::bg_pixbuf = $::bg_pixbuf_orig->scale_simple($w, $h, 'bilinear');
+	my $rect = Gtk2::Gdk::Rectangle->new(0, 0, $w, $h);
+	$gdk_window->invalidate_rect($rect, TRUE);
+    }
+    $gdk_window->draw_pixbuf($widget->style->bg_gc('normal'), $::bg_pixbuf, 0, 0, 0, 0, $w, $h, 'none', 0, 0);
+    return 1;
+}
+
+my $bg_window;
+
+sub show_bg_window {
+    $bg_window = Gtk2::Window->new();
+    $bg_window->signal_connect('destroy', sub { Gtk2->main_quit; });
+    $bg_window->maximize;
+    $bg_window->set_keep_below(TRUE);
+    $bg_window->signal_connect('expose-event', \&draw_bg_pixbuf);
+    $bg_window->show;
+}
+
+sub hide_bg_window {
+    $bg_window->hide;
+    $bg_window->destroy;
+}
+
+1;
+
diff -Nur drakxtools-14.41.old/perl-install/mygtk2.pm drakxtools-14.41/perl-install/mygtk2.pm
--- drakxtools-14.41.old/perl-install/mygtk2.pm	2013-04-16 17:25:43.000000000 +0400
+++ drakxtools-14.41/perl-install/mygtk2.pm	2013-04-18 23:29:56.084012363 +0400
@@ -1430,8 +1430,9 @@
 my @icon_paths;
 sub add_icon_path { push @icon_paths, @_ }
 sub _icon_paths() {
-   (@icon_paths, (exists $ENV{SHARE_PATH} ? ($ENV{SHARE_PATH}, "$ENV{SHARE_PATH}/icons", "$ENV{SHARE_PATH}/libDrakX/pixmaps") : ()),
-    "/usr/lib/libDrakX/icons", "pixmaps", 'data/icons', 'data/pixmaps', 'standalone/icons', '/usr/share/rpmdrake/icons');
+    my $loc = (($ENV{'LC_MESSAGES'} =~ m/ru_RU/) ? 'ru' : 'en');
+    (@icon_paths, (exists $ENV{SHARE_PATH} ? ($ENV{SHARE_PATH}, "$ENV{SHARE_PATH}/icons", "$ENV{SHARE_PATH}/libDrakX/pixmaps/$loc", "$ENV{SHARE_PATH}/libDrakX/pixmaps") : ()),
+    "/usr/lib/libDrakX/icons", "pixmaps/$loc", "pixmaps", 'data/icons', 'data/pixmaps', 'standalone/icons', '/usr/share/rpmdrake/icons');
 }  
 
 sub main {
