diff -Naur drakxtools-14.41/perl-install/standalone/fileshareset drakxtools-14.41.tpg/perl-install/standalone/fileshareset
--- drakxtools-14.41/perl-install/standalone/fileshareset	2013-04-16 13:25:43.000000000 +0000
+++ drakxtools-14.41.tpg/perl-install/standalone/fileshareset	2014-03-05 19:24:07.706700130 +0000
@@ -396,11 +396,11 @@
 	if (system('PATH=/bin:/sbin pidof smbd >/dev/null') != 0 ||
 	    system('PATH=/bin:/sbin pidof nmbd >/dev/null') != 0) {
 	    # trying to start the server...
-	    my ($f) = grep { -f $_ } map { "/etc/init.d/$_" } 'smb', 'samba', 'rc.samba';
+	    my ($f) = grep { -f $_ } map { "/lib/systemd/system/$_" } 'smb.service', 'nmb.service';
 	    if ($f) {
-		system($f, $_) foreach 'stop', 'start';
+		system("systemctl",$_,File::Basename::basename($f)) foreach 'stop', 'start';
 	    } else {
-		print STDERR "Error: Can't find the samba init script \n";
+		print STDERR "Error: Can't find the samba service file \n";
 	    }
 	}
 	exit 0;
