diff -c -P -r drakxtools-13.51.old/perl-install/bootloader.pm drakxtools-13.51/perl-install/bootloader.pm
*** drakxtools-13.51.old/perl-install/bootloader.pm	2011-06-27 19:02:32.000000000 +0400
--- drakxtools-13.51/perl-install/bootloader.pm	2011-08-17 10:04:19.000000000 +0400
***************
*** 1943,1951 ****
      if ($main_method eq 'grub' || $main_method eq 'lilo') {
  	$do_pkgs->ensure_binary_is_installed($main_method, $main_method, 1) or return 0;
  	if ($bootloader->{method} eq 'grub-graphic') {
! 	    $do_pkgs->ensure_is_installed('mandriva-gfxboot-theme', '/usr/share/gfxboot/themes/Mandriva/boot/message', 1) or return 0;
! 	}
!     }
      1;
  }
  
--- 1943,1988 ----
      if ($main_method eq 'grub' || $main_method eq 'lilo') {
  	$do_pkgs->ensure_binary_is_installed($main_method, $main_method, 1) or return 0;
  	if ($bootloader->{method} eq 'grub-graphic') {
! #check installation gfxboot-theme
! 	    $pkg_install = 0;
! 	    opendir GFX_LIST, "$::prefix/usr/share/gfxboot/themes";
! 	    @names = readdir GFX_LIST;
! 	    foreach $name (@names){
! 		if(-e "$::prefix/usr/share/gfxboot/themes/$name/boot/message"){
! 		    $pkg_install = 1;
! 		    last;
! 		};
! 	    };
! 	    closedir GFX_LIST;
! 	    if(!$pkg_install){
! 		open INST_LIST,"urpmi_rpm-find-leaves -f |";
! 		while(<INST_LIST>){
! 		    @packages = readline INST_LIST;
! 		    foreach $package (@packages){
! 			@param = split / /, $package;
! 			if(@param[0] =~ m/gfxboot-theme/){
! 			    $pkg_install = 1;
! 			    last;
! 			}
! 		    }
! 		    if($pkg_install){
! 			last;
! 		    }
! 		};
! 		close INST_LIST;
! 	    };
! #gfxboot-theme not found
! 	    if(!$pkg_install){
! 		open(FD_RELEASE_NAME,'/etc/mandriva-release');
! 		$release_name = readline(FD_RELEASE_NAME);
! 		if($release_name =~ m/rosa/i){
! 		    $do_pkgs->ensure_is_installed('rosa-gfxboot-theme', '/usr/share/gfxboot/themes/ROSA/boot/message', 1) or return 0;
! 		}else{
! 		    $do_pkgs->ensure_is_installed('mandriva-gfxboot-theme', '/usr/share/gfxboot/themes/Mandriva/boot/message', 1) or return 0;
! 		};
! 	    };
! 	};
!     };
      1;
  }
  
